.github/workflows/docs.yml

```diff
@@ -0,0 +1,19 @@
+name: docs
```

Comment:
Can you test out this workflow on a fork first?  I'd like to see what the docs being pushed out by this workflow look like before merging

I'm also not convinced this is quite right - we're pushing to a 'main' subfolder in gh-pages (to match eventual sphinx-multiversion support), and I think we might need dependencies like TF/PT installed first to generate the API documentation.